### PR TITLE
Prevent Warnings from `Lazy_String` on PHP 8.1

### PR DESCRIPTION
**Pull request by bordoni**
Resolves #1823

### CODE REVIEW

File: src/Tribe/Utils/Lazy_String.php [hunk 1]
```diff
@@ -71,7 +71,12 @@ public function __construct( callable $callback, $escape_callback = 'esc_html' )
 	 */
 	public function __toString() {
```
Comment (Camwyn) [inline src/Tribe/Utils/Lazy_String.php:72]:
```suggestion
	public function __toString(): string {
```

File: src/Tribe/Utils/Lazy_String.php [hunk 2]
```diff
@@ -71,7 +71,12 @@ public function __construct( callable $callback, $escape_callback = 'esc_html' )
...
 		if ( null === $this->string ) {
-			$this->string = call_user_func( $this->value_callback );
+			$value =  call_user_func( $this->value_callback );
+			if ( ! $value instanceof \Generator ) {
```
Comment (Camwyn) [inline src/Tribe/Utils/Lazy_String.php:75]: Why `Generator`?

**Review by Camwyn — CHANGES_REQUESTED** [review]
We should enforce via the return cast.

File: src/Tribe/Utils/Lazy_String.php (hunk 2 above)
Comment (bordoni) [reply src/Tribe/Utils/Lazy_String.php:75]: Because if we use `yield` it will return that.

**Review by Camwyn — APPROVED** [review]

**Comment by tyrann0us** [comment]
Hi @bordoni, we tried to update the Events Calendar to `v6` where this fix is included. However, multiple issues made us wait for a more stable 6.* version.

Is there any chance to backport this fix to 5.*? Thanks!

**Pull request merged**